Src/CSharpier/CommandLineFormatter.cs

```diff
@@ -0,0 +1,323 @@
+            }
+            catch (Exception ex)
+            {
+                Interlocked.Increment(ref this.Files);
+                Console.WriteLine(
+                    GetPath() + " - threw exception while formatting"
+                );
+                Console.WriteLine(ex.Message);
+                Console.WriteLine(ex.StackTrace);
+                Console.WriteLine();
+                Interlocked.Increment(ref this.ExceptionsFormatting);
+                return;
+            }
+
+            if (result.Errors.Any())
+            {
+                Interlocked.Increment(ref this.Files);
+                Console.WriteLine(GetPath() + " - failed to compile");
+                return;
+            }
+
+            if (!result.FailureMessage.IsBlank())
+            {
+                Interlocked.Increment(ref this.Files);
+                Console.WriteLine(GetPath() + " - " + result.FailureMessage);
+                return;
+            }
+
+            if (this.Validate)
+            {
+                var syntaxNodeComparer = new SyntaxNodeComparer(
+                    fileReaderResult.FileContents,
+                    result.Code,
+                    cancellationToken
+                );
+
+                try
+                {
+                    var failure =
+                        await syntaxNodeComparer.CompareSourceAsync(
+                            cancellationToken
+                        );
+                    if (!string.IsNullOrEmpty(failure))
+                    {
+                        Interlocked.Increment(
+                            ref this.FailedSyntaxTreeValidation
+                        );
+                        Console.WriteLine(
+                            GetPath() + " - failed syntax tree validation"
+                        );
+                        Console.WriteLine(failure);
+                    }
+                }
+                catch (Exception ex)
+                {
+                    Interlocked.Increment(ref this.ExceptionsValidatingSource);
+                    Console.WriteLine(
+                        GetPath()
+                        + " - failed with exception during syntax tree validation"
+                        + Environment.NewLine
+                        + ex.Message
+                        + ex.StackTrace
+                    );
+                }
+            }
+
+            if (this.Check)
+            {
+                if (result.Code != fileReaderResult.FileContents)
+                {
+                    Console.WriteLine(GetPath() + " - was not formatted");
+                    Interlocked.Increment(ref this.UnformattedFiles);
+                }
+            }
+
+            cancellationToken.ThrowIfCancellationRequested();
+            Interlocked.Increment(ref this.Files);
+
+            if (!this.Check)
+            {
+                // purposely avoid async here, that way the file completely writes if the process gets cancelled while running.
```

Comment:
Do you know if that's true?

Reply:
Yeah. Not every time, but it probably happened every other time I cancelled the process when I was testing.